subsys/modem/modem_cmux.c

```diff
@@ -228,16 +228,15 @@ static uint16_t modem_cmux_transmit_frame(struct modem_cmux *cmux,
 	ring_buf_put(&cmux->transmit_rb, &byte, 1);
 
 	/* Data length */
-	if (data_len > 127) {
-		byte = data_len << 1;
+	byte = data_len << 1;
+	if (data_len & ~0x7f) {
 		fcs = crc8(&byte, 1, MODEM_CMUX_FCS_POLYNOMIAL, fcs, true);
 		ring_buf_put(&cmux->transmit_rb, &byte, 1);
-		byte = 0x01 | (data_len >> 7);
-		ring_buf_put(&cmux->transmit_rb, &byte, 1);
+		byte = data_len >> 7;
 	} else {
-		byte = 0x01 | (data_len << 1);
-		ring_buf_put(&cmux->transmit_rb, &byte, 1);
+		byte |= 0x1; /* EA bit to indicate that the length field is one byte long. */
 	}
+	ring_buf_put(&cmux->transmit_rb, &byte, 1);
```

Comment:
You are correct regarding the error :) However, the changes you are making here are less readable than the code you are replacing (especially `data_len & ~0x7f` instead of `data_len > 127`) :(

You need only change https://github.com/zephyrproject-rtos/zephyr/blob/c05a483ba4251589b713a1231fb6ad4c75cebbaa/subsys/modem/modem_cmux.c#L235C1-L235C33
to
```
byte = data_len >> 7;
```